horovod/common/operations.cc

```diff
@@ -819,6 +1080,15 @@ void BackgroundThreadLoop(HorovodGlobalState& state, MPIContext& ctx) {
                                        true);
   }
 
+  // Override response cache capacity, if it's set.
+  // TODO: Put in param manager?
```

Comment:
So I made a first pass in adding this to param manager, but it would be nice if @tgaddair could review and make sure things look alright. 

Before this though, I think there may be a bug in the current master branch with the autotuner. I get `std::bad_alloc` exceptions thrown when the autotuner is active after a few hundred steps (seems to occur when the parameters are updated). This was with a TF RN50 case with 2 processes. Can one of you verify that the autotuner is still in a working state?

Reply:
If it's useful, reverting commit aa605d6de1e07d1184e8793b6d4f65cee9d4efb3 seems to fix the exception issue.

Reply:
The changes look good to me.  I haven't seen the autotuner issue you're describing before, but I'll play around with a few examples and see if I can reproduce it.

Reply:
@tgaddair I am able to produce the error using the `tensorflow_synthetic_benchmark.py` file with Horovod 0.16.1 installed with pip. Running with the command `mpirun -np 2 -x HOROVOD_AUTOTUNE=1 -x HOROVOD_AUTOTUNE_LOG=at.log python tensorflow_synthetic_benchmark.py --num-iters 300` generates the exception to after reporting iteration 23. This doesn't occur with Horovod 0.16.0.

Reply:
Let me see if I can reproduce and get back to you.